ssl/ssl_stat.c

```diff
@@ -343,7 +343,7 @@ const char *SSL_alert_desc_string_long(int value)
     case SSL3_AD_CLOSE_NOTIFY:
         return "close notify";
     case SSL3_AD_UNEXPECTED_MESSAGE:
-        return "unexpected_message";
+        return "unexpected message";
```

Comment:
This looks like something that a test script might check for, so not sure this one should be changed

Reply:
return result here is just a description string, there is no reason that other switch cases returning string without underline, but "unexpected_message" is exceptional, so I believe that it must be a typo!

Reply:
the tls13hrr test looks for an unexpected message error, though I can't tell if it checks for the string or the numerical error code.  Does this pass the unit tests for you locally?

Additionally, This gets printed when ssl tracing is enabled, and I'm not sure what other user scripts are out in the wild that expect the underscore to be there for whatever greps they might be doing.

I'll approve the unit tests here to see how the run goes

Reply:
Given that it's not clear who might be using this, and that it's not a user-facing error message, or comment, or documentation, and that the only thing wrong with it is the underscore, I'd be much happier not changing this one. The other fix is perfectly fine

Reply:
> the tls13hrr test looks for an unexpected message error, though I can't tell if it checks for the string or the numerical error code. Does this pass the unit tests for you locally?
> 
> Additionally, This gets printed when ssl tracing is enabled, and I'm not sure what other user scripts are out in the wild that expect the underscore to be there for whatever greps they might be doing.
> 
> I'll approve the unit tests here to see how the run goes

Yeah, you're right, test case in test/recipes/70-test_tls13hrr.t checks for an unexpected message, but checking the numerical form, not the string form

Reply:
The point is we can't know if anyone else is using this, not just within this repo, so safest not to change it

Reply:
> Given that it's not clear who might be using this, and that it's not a user-facing error message, or comment, or documentation, and that the only thing wrong with it is the underscore, I'd be much happier not changing this one. The other fix is perfectly fine

Let's see the doc of [SSL_alert_desc_string_long](https://www.openssl.org/docs/manmaster/man3/SSL_alert_desc_string_long.html) in master branch. It's quite clear that it will return "unexpected message" and when people using this function, they normally will consult the doc! Currently, the doc and the result returned does not match, if we don't merge this change, then we should change the doc as "unexpected_message", which will break the code style, since other return values are without the underline.

![unexpected_message](https://github.com/openssl/openssl/assets/11711429/90b1d2c2-63ae-4b5e-8a6d-3705db4e1a19)